src/main/java/org/thymeleaf/standard/expression/SwitchCaseExpression.java

```diff
@@ -0,0 +1,167 @@
+
+	private final Expression switchConditionExpression;
+
+	private final CaseSequence caseSequence;
+
+	public SwitchCaseExpression(final Expression switchConditionExpression, final CaseSequence cases) {
+		super();
+		Validate.notNull(switchConditionExpression, "Switch condition cannot be null");
+		this.switchConditionExpression = switchConditionExpression;
+		this.caseSequence = cases;
+	}
+
+	public boolean hasCases() {
+		return this.caseSequence != null && this.caseSequence.size() > 0;
+	}
+
+	@Override
+	public String getStringRepresentation() {
+		final StringBuilder sb = new StringBuilder();
+
+		if (this.switchConditionExpression instanceof ComplexExpression) {
+			sb.append(Expression.NESTING_START_CHAR);
+			sb.append(this.switchConditionExpression);
+			sb.append(Expression.NESTING_END_CHAR);
+		} else {
+			sb.append(this.switchConditionExpression);
+		}
+
+		sb.append(OPERATOR);
+
+		if (this.hasCases()) {
+			sb.append(this.caseSequence.getStringRepresentation());
+		}
+		return sb.toString();
+	}
+
+	public Expression getSwitchConditionExpression() {
+		return this.switchConditionExpression;
+	}
+
+	public CaseSequence getCases() {
+		return caseSequence;
+	}
+
+	public static ExpressionParsingState composeSwitchCaseExpression(final ExpressionParsingState state,
+			final int nodeIndex) {
+
+		final String input = state.get(nodeIndex).getInput();
+		if (StringUtils.isEmptyOrWhitespace(input)) {
+			return null;
+		}
+
+		// Trying to fail quickly...
+		final int operatorPos = input.indexOf(OPERATOR);
+		if (operatorPos == -1) {
+			return state;
+		}
+
+		final String switchStr = input.substring(0, operatorPos).trim();
+		final String remainder = input.substring(operatorPos + 1).trim();
+
+		// Check the switch condition and fail quickly if it is not valid
+		final Expression switchExpr = ExpressionParsingUtil.parseAndCompose(state, switchStr);
+		if (switchExpr == null) {
+			return null;
+		}
+
+		// remainder string should represent the case sequence
+		// condition: at least 1 case should exist
+
+		CaseSequence sequence = CaseUtils.composeSequence(remainder);
+
+		if (sequence == null) {
+			return null;
+		}
+
+		final SwitchCaseExpression switchCaseExpressionResult = new SwitchCaseExpression(switchExpr, sequence);
+		state.setNode(nodeIndex, switchCaseExpressionResult);
+
+		return state;
+	}
+
+	public static Object executeSwitchCase(IExpressionContext context, SwitchCaseExpression expression,
+			StandardExpressionExecutionContext expContext) {
+
+		if (logger.isTraceEnabled()) {
+			logger.trace("[THYMELEAF][{}] Evaluating switch expression: \"{}\"", TemplateEngine.threadIndex(),
+					expression.getStringRepresentation());
+		}
+
+		// obtain switch condition
+		final Object switchObj = expression.getSwitchConditionExpression().execute(context, expContext);
+		Object defaultCase = null;
+
+		// compare each case 'key' with switch variable
+		// if the switch variable matches on of the cases, return its value
```

Comment:
@danielfernandez Sorry, I didn't wanted to harm no one. :). What I wanted to say was that the code speaks for itself  so adding extra comment on a nice written code is useless. I didn't wanted to be a jerk at all.